### PR TITLE
If user provides wrong credentials will verify and re-ask if invalid during setting config

### PR DESCRIPTION
**Pull request by AnirudhPokala**
* Added the validation of credentials during config setting ( config, check-node, prep-node command), and will re-run the config setting if not entered valid user credentials. 
* Removed the redundancy of authentication token generation during check-node, and created a new argument auth for checkNode function.
Screenshot 2021-02-26 at 8 03 53 PM

### CODE REVIEW

File: cmd/checkNode.go [hunk 1]
```diff
@@ -32,27 +33,53 @@ func init() {
...
-	if err != nil {
-		zap.S().Fatalf("Unable to load clients needed for the Cmd. Error: %s", err.Error())
-	}
+	var (
+		ctx  pmk.Config
+		c    pmk.Client
+		err  error
+		auth keystone.KeystoneAuth
+		flag = true
+	)
```
Comment (anupbarve) [inline cmd/checkNode.go:43]: Please put a comment around what each of these fields indicates.

File: cmd/checkNode.go [hunk 2]
```diff
@@ -32,27 +33,53 @@ func init() {
...
+		zap.S().Debug("==========Validating the User Credentials==========")
+
+		auth, err = c.Keystone.GetAuth(
+			ctx.Username,
+			ctx.Password,
+			ctx.Tenant,
+		)
+
+		if err != nil {
+			zap.S().Debug("Invalid credentials entered (Username/Password/Tenant)")
```
Comment (anupbarve) [inline cmd/checkNode.go:71]: Could you also put this case in the description. It is not clear to me whether a user will get this error on the console ? From the looks of it, the user won't get any warning/error on the console.

File: cmd/config.go [hunk 3]
```diff
@@ -23,7 +23,43 @@ var configCmdCreate = &cobra.Command{
 func configCmdCreateRun(cmd *cobra.Command, args []string) {
 	zap.S().Debug("==========Running set config==========")
 	// invoked the configcreate command from pkg/pmk
-	if err := pmk.StoreConfig(pmk.ConfigCmdCreateRun(), util.Pf9DBLoc); err != nil {
+
+	var (
+		flag = true
+		ctx  pmk.Config
```
Comment (anupbarve) [inline cmd/config.go:29]: Need a good name for flag, add some comment.

File: cmd/checkNode.go (hunk 1 above)
Comment (anupbarve) [reply cmd/checkNode.go:43]: Also, need a more intuitive name for `flag`.

File: cmd/prepNode.go [hunk 4]
```diff
@@ -46,26 +47,57 @@ func init() {
...
+		)
+
+		if err != nil {
+			zap.S().Debug("Invalid credentials entered (Username/Password/Tenant)")
+		} else {
+			if err := pmk.StoreConfig(ctx, util.Pf9DBLoc); err != nil {
+				zap.S().Errorf("Failed to store config: %s", err.Error())
+			}
+			flag = false
+		}
```
Comment (anupbarve) [inline cmd/prepNode.go:91]: I remember that we had discussed about getting this redundant code in a common function.
What happened to that ? Did you hit any roadblocks ?

File: pkg/pmk/config.go [hunk 5]
```diff
@@ -121,6 +121,41 @@ func ConfigCmdCreateRun() Config {
...
+	if err != nil {
+		zap.S().Fatalf("Invalid credentials(): %s", err.Error())
+	} else {
+		if err := StoreConfig(ctx, util.Pf9DBLoc); err != nil {
+			zap.S().Errorf("Failed to store config: %s", err.Error())
+		}
+	}
+   return ctx, c
 }
+*/
```
Comment (anupbarve) [inline pkg/pmk/config.go:161]: Remove or (Enable and reuse) :)

**Review by anupbarve — CHANGES_REQUESTED** [review]

File: pkg/pmk/config.go (hunk 5 above)
Comment (AnirudhPokala) [reply pkg/pmk/config.go:161]: sure anup will remove, its not working actually, i will discuss during standup

File: cmd/checkNode.go (hunk 2 above)
Comment (AnirudhPokala) [reply cmd/checkNode.go:71]: Ya anup, actually its getting into log so changed it now to get in on the console if entered invalid credentials.

File: cmd/config.go (hunk 3 above)
Comment (AnirudhPokala) [reply cmd/config.go:29]: Ya modified to credentialsFlag.

File: cmd/prepNode.go (hunk 4 above)
Comment (AnirudhPokala) [reply cmd/prepNode.go]: Ya anup, i had tried but there was an issue, will discuss with you.

File: cmd/checkNode.go (hunk 1 above)
Comment (AnirudhPokala) [reply cmd/checkNode.go]: Ya done anup had put the comments where needed.

File: cmd/prepNode.go (hunk 4 above)
Comment (AnirudhPokala) [reply cmd/prepNode.go]: Ya anup wrapped up the loading and validating into a function.

**Comment by AnirudhPokala** [comment]
Due to many changes made after this PR, it was creating too many merge conflicts. So raised a new PR.

**Pull request closed without merging**